### ISSUE TITLE
Move DTOs from the service to the rest layer

### ISSUE THREAD

**Issue by mshima** [jhipster/generator-jhipster#21821]
##### **Overview of the feature request**

JHipster DTOs are the rest api definitions and are probably best fit in the rest layer.

The main benefit is to simplify the templates by moving all dto operations to the controller.

##### **Related issues or PR**

- [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by gzsombor** [comment]
That wont work, whenever you start to deal with lazily initialized JPA entities, because you need to that in a transaction.

**Comment by mshima** [comment]
We are using optimized eager loading in the repository layer.
And AFAIK we support DTOs without services. In this case DTO looks orphan without the corresponding service.

**Comment by Tcharl** [comment]
Nice idea, dto at service level cause issues for complex custom service relying on others: there are many cases where you get  the DTO, retransform to entity to get a field that is used by the other service, ...
Just to think about this custom (and generated) services and lazyloadingexceptions (note that I'm not a fan of the current eager fetch neither, would have preferred a projected view but it's another topic)

**Comment by deepu105** [comment]
In that case should we remove the VO objects on the Rest layer as well as they are just a mirror of DTOs. We created them to avoid leaking transactional layers to cleint side AFAIK

**Comment by pascalgrimaud** [comment]
I agree with @gzsombor 

It will probably work for all simple generated projects.
It won't work for real and existing projects.

DTOs support without service has always been a bad idea.
As @deepu105 said, the VO objects were created for that reason.

It's a really major breaking changes and we should collect more feedbacks before doing this change.
Just my opinion :-)

cc @jhipster/developers

**Comment by jdubois** [comment]
Indeed, we have *optional* DTOs from the repository to service layers, for those who don't like to have their JPA entities crossing all layers (which isn't my case, I love to do this!!). And then the VOs, for "view objects", are from the service to Web layers - they are similar to the view objects we also have in the front end.
The DTOs are for the business layer, and the VOs are for the view layer.

**Comment by deepu105** [comment]
> In that case should we remove the VO objects on the Rest layer as well as they are just a mirror of DTOs. We created them to avoid leaking transactional layers to cleint side AFAIK

Also I thought we disallowed DTO without service. Atleast thats how I remember it, or maybe its only on the JDL side, i need to check the code to be sure

**Comment by mshima** [comment]
I suppose this needs more discussion so I will remove from v8 tasks.

**Comment by mshima** [comment]
@DanielFran @mraible @qmonmert should we close this?

**Comment by Abu1982** [comment]
Hi, I'd like to work on this issue.

I've reviewed the thread and confirmed there is no active PR for it.

**Planned approach:** Analyze and fix based on error signals in jhipster/generator-jhipster

I'll submit a focused PR once the fix is ready and locally verified.
